core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java

```diff
@@ -68,31 +70,54 @@
   private RestoredDlpItemService restoredDlpItemService;
 
   private DlpOperationProcessor dlpOperationProcessor;
+  private TrashService          trashService;
 
   private LinkManager linkManager;
 
   public FileDlpConnector(InitParams initParams, FileSearchServiceConnector fileSearchServiceConnector,
-                          RepositoryService repositoryService, IndexingService indexingService, DlpOperationProcessor dlpOperationProcessor, RestoredDlpItemService restoredDlpItemService, LinkManager linkManager) {
+                          RepositoryService repositoryService, IndexingService indexingService,
+                          DlpOperationProcessor dlpOperationProcessor, RestoredDlpItemService restoredDlpItemService, LinkManager linkManager,
+                          TrashService trashService) {
     super(initParams);
     this.repositoryService = repositoryService;
     this.indexingService = indexingService;
     this.fileSearchServiceConnector = fileSearchServiceConnector;
     this.restoredDlpItemService = restoredDlpItemService;
     this.dlpOperationProcessor = dlpOperationProcessor;
     this.linkManager = linkManager;
+    this.trashService=trashService;
   }
 
   @Override
   public boolean processItem(String entityId) {
     LOGGER.debug("Process item {}",entityId);
+    if (isInTrash(entityId)) {
+      //if a document is in trash, we cannot check it because it is not indexed.
+      //so we return true to remove it from the queue
+      return true;
+    }
     if (!isIndexedByEs(entityId) || editorOpened(entityId)) {
       return false;
     } else {
       checkMatchKeywordAndTreatItem(entityId);
     }
     return true;
   }
-
+  
+  private boolean isInTrash(String entityId) {
+    ExtendedSession session = null;
+    try {
+      session = (ExtendedSession) WCMCoreUtils.getSystemSessionProvider().getSession(COLLABORATION_WS, repositoryService.getCurrentRepository());
+      Node node = session.getNodeByIdentifier(entityId);
+      boolean result = trashService.isInTrash(node);
+      LOGGER.debug("Item {} isEditorOpened={}", entityId, result);
```

Comment:
I think you need to update the log message

Reply:
doen, thanks @ahamdi